pkg/operator/controllers/checkers/ingresscertificatechecker/checker.go

```diff
@@ -59,13 +62,41 @@ func (r *checker) Check(ctx context.Context) error {
 		return err
 	}
 
-	if ingress.Spec.DefaultCertificate == nil {
+	cluster, err := r.arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	return validateCertificate(cv.Spec.ClusterID, ingress.Spec.DefaultCertificate, cluster.Spec.Domain)
+}
+
+func validateCertificate(clusterId configv1.ClusterID, certificate *corev1.LocalObjectReference, clusterDomain string) error {
+	clusterHasManagedDomain := clusterHasManagedDomain(clusterDomain)
+
+	if certificate == nil {
+		if clusterHasManagedDomain {
+			return fmt.Errorf("ARO has not provisioned the default ingress certificate because it is empty and cluster has a managed domain")
+		}
 		return errNoDefaultCertificate
 	}
 
-	if ingress.Spec.DefaultCertificate.Name != string(cv.Spec.ClusterID)+"-ingress" {
-		return fmt.Errorf("custom ingress certificate in use: %q", ingress.Spec.DefaultCertificate.Name)
+	certificateName := certificate.Name
+	certificateNameIsValid := certificateNameIsValid(certificateName, clusterId)
+
+	if clusterHasManagedDomain && !certificateNameIsValid {
+		return fmt.Errorf("custom ingress certificate in use: %q", certificateName)
```

Comment:
Not sure if we should log an error message with the certificate name. That's not specifically secret, but the name of the cert does not really matter, as long as it is not the ARO provisionned one

Reply:
Good point. It was already like this before this PR. Changing it now. 
Tks!